paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1150,6 +1156,30 @@ def paasta_spark_run(args):
         needs_docker_cfg=needs_docker_cfg,
         auto_set_temporary_credentials_provider=auto_set_temporary_credentials_provider,
     )
+
+    # Mass enable Spark DRA. This is to enable Dynamic Resource Allocation in Spark through executor ranges.
+    # If the number of executor instances of a Spark job lie in the specified range, dynamicAllocation configs
+    # will be added. More info: y/spark-dra
+    # TODO: To be removed when DRA is enabled by default
+    if (
+        args.cluster_manager == CLUSTER_MANAGER_K8S
+        and "spark.dynamicAllocation.enabled" not in spark_conf
+    ):
+        num_executors = int(spark_conf["spark.executor.instances"])
+        for exec_range in EXECUTOR_RANGES:
+            if exec_range[0] <= num_executors <= exec_range[1]:
+                log.warn(
+                    PaastaColors.yellow(
+                        "Spark Dynamic Resource Allocation (DRA) enabled for this batch. "
+                        "More info: y/spark-dra. If your job is performing worse because "
+                        "of DRA, consider disabling DRA. To disable, please provide "
+                        "spark.dynamicAllocation.enabled=false in --spark-args\n"
+                    )
+                )
+                spark_conf["spark.dynamicAllocation.enabled"] = "true"
+                spark_conf = get_dra_configs(spark_conf)
+                break
```

Comment:
suggestion: it'd be nice to stuff this into a function so that cleanup is easier later, but up to y'all!

Reply:
Makes sense. Moved to a function